cdc/log.cc

```diff
@@ -1327,13 +1329,15 @@ cdc::cdc_service::impl::augment_mutation_call(lowres_clock::time_point timeout,
                 tracing::trace(tr_state, "CDC: Preimage not enabled for the table, not querying current value of {}", m.decorated_key());
             }
 
-            return f.then([trans = std::move(trans), &mutations, idx, tr_state = std::move(tr_state), &stats] (lw_shared_ptr<cql3::untyped_result_set> rs) mutable {
+            return f.then([trans = std::move(trans), &mutations, idx, tr_state = std::move(tr_state), &stats, &meta] (lw_shared_ptr<cql3::untyped_result_set> rs) mutable {
                 auto& m = mutations[idx];
                 auto& s = m.schema();
                 tracing::trace(tr_state, "CDC: Generating log mutations for {}", m.decorated_key());
                 int generated_count;
+                size_t size_before_appends = mutations.size();
+                const bool do_split = should_split(m, *s);
                 stats::part_type_set touched_parts;
-                if (should_split(m, *s)) {
+                if (do_split) {
                     stats.counters_total.split_count++;
```

Comment:
Do we still need this? Can't generate_range take care of this as well?

Reply:
Now both "failed" and "total" metrics are updated after all cdc mutation writes finish.